File: app/src/test/java/com/mgaetan89/showsrage/model/ShowsFilters_Status_IsTest.java
package com.mgaetan89.showsrage.model;

import com.mgaetan89.showsrage.model.ShowsFilters.Status;

import org.junit.Test;
import org.junit.runner.RunWith;
import org.junit.runners.Parameterized;

import java.util.Arrays;
import java.util.Collection;

import static org.assertj.core.api.Assertions.assertThat;

@RunWith(Parameterized.class)
public class ShowsFilters_Status_IsTest {
	@Parameterized.Parameter(1)
	public boolean all;

	@Parameterized.Parameter(2)
	public boolean continuing;

	@Parameterized.Parameter(3)
	public boolean ended;

	@Parameterized.Parameter(0)
	public int status;

	@Parameterized.Parameter(4)
	public boolean unknown;

	@Test
	public void isAll() {
		assertThat(ShowsFilters.Status.Companion.isAll(this.status)).isEqualTo(this.all);
	}

	@Test
	public void isContinuing() {
		assertThat(ShowsFilters.Status.Companion.isContinuing(this.status)).isEqualTo(this.continuing);
	}

	@Test
	public void isEnded() {
		assertThat(ShowsFilters.Status.Companion.isEnded(this.status)).isEqualTo(this.ended);
	}

	@Test
	public void isUnknown() {
		assertThat(ShowsFilters.Status.Companion.isUnknown(this.status)).isEqualTo(this.unknown);
	}

	@Parameterized.Parameters
	public static Collection<Object[]> data() {
		int all = Status.ALL.getStatus();
		int continuing = Status.CONTINUING.getStatus();
		int ended = Status.ENDED.getStatus();
		int unknown = Status.UNKNOWN.getStatus();

		return Arrays.asList(new Object[][]{
				{0, false, false, false, false},
				{all, true, true, true, true},
				{continuing, false, true, false, false},
				{continuing | ended, false, true, true, false},
				{continuing | ended | unknown, true, true, true, true},
				{continuing | unknown, false, true, false, true},
				{ended, false, false, true, false},
				{ended | unknown, false, false, true, true},
				{unknown, false, false, false, true},
		});
	}
}

File: app/src/test/kotlin/com/mgaetan89/showsrage/model/ShowsFilters_Status_IsTest.kt
package com.mgaetan89.showsrage.model

import com.mgaetan89.showsrage.model.ShowsFilters.Status
import org.assertj.core.api.Assertions.assertThat
import org.junit.Test
import org.junit.runner.RunWith
import org.junit.runners.Parameterized

@RunWith(Parameterized::class)
class ShowsFilters_Status_IsTest {
    @Parameterized.Parameter(1)
    var all: Boolean = false

    @Parameterized.Parameter(2)
    var continuing: Boolean = false

    @Parameterized.Parameter(3)
    var ended: Boolean = false

    @Parameterized.Parameter(0)
    var status: Int = 0

    @Parameterized.Parameter(4)
    var unknown: Boolean = false

    @Test
    fun isAll() {
        assertThat(ShowsFilters.Status.isAll(this.status)).isEqualTo(this.all)
    }

    @Test
    fun isContinuing() {
        assertThat(ShowsFilters.Status.isContinuing(this.status)).isEqualTo(this.continuing)
    }

    @Test
    fun isEnded() {
        assertThat(ShowsFilters.Status.isEnded(this.status)).isEqualTo(this.ended)
    }

    @Test
    fun isUnknown() {
        assertThat(ShowsFilters.Status.isUnknown(this.status)).isEqualTo(this.unknown)
    }

    companion object {
        @JvmStatic
        @Parameterized.Parameters
        fun data(): Collection<Array<Any>> {
            val all = Status.ALL.status
            val continuing = Status.CONTINUING.status
            val ended = Status.ENDED.status
            val unknown = Status.UNKNOWN.status

            return listOf(
                    arrayOf(0, false, false, false, false),
                    arrayOf(all, true, true, true, true),
                    arrayOf(continuing, false, true, false, false),
                    arrayOf(continuing or ended, false, true, true, false),
                    arrayOf(continuing or ended or unknown, true, true, true, true),
                    arrayOf(continuing or unknown, false, true, false, true),
                    arrayOf(ended, false, false, true, false),
                    arrayOf(ended or unknown, false, false, true, true),
                    arrayOf(unknown, false, false, false, true)
            )
        }
    }
}
